File: backend/src/main/java/com/tayyar/order/OrderStatus.java
package com.tayyar.order;

public enum OrderStatus {
    PENDING_PAYMENT,
    PLACED,
    ACCEPTED,
    PREPARING,
    READY_FOR_PICKUP,
    OUT_FOR_DELIVERY,
    DELIVERED,
    PAYMENT_FAILED,
    REJECTED,
    CANCELLED
}

File: frontend/src/utils/orderStatus.ts
import type { OrderStatus } from '../api/contracts'

export const orderStatusLabels: Record<OrderStatus, string> = {
  PENDING_PAYMENT: 'Pending payment', PLACED: 'Placed', ACCEPTED: 'Accepted', PREPARING: 'Preparing',
  READY_FOR_PICKUP: 'Ready for pickup', OUT_FOR_DELIVERY: 'Out for delivery', DELIVERED: 'Delivered',
  PAYMENT_FAILED: 'Payment failed', REJECTED: 'Rejected', CANCELLED: 'Cancelled',
}

const terminal = new Set<OrderStatus>(['DELIVERED', 'PAYMENT_FAILED', 'REJECTED', 'CANCELLED'])
export const statusLabel = (status: OrderStatus) => orderStatusLabels[status]
export const isTerminalOrderStatus = (status: OrderStatus) => terminal.has(status)
